torch/distributed/_tensor/_xla.py

```diff
@@ -197,4 +198,75 @@ def xla_distribute_module(
     input_fn: Optional[Callable[..., None]] = None,
     output_fn: Optional[Callable[..., None]] = None,
 ) -> nn.Module:
-    raise NotImplementedError
+    """
+    This function annotates all module parameters for auto-partitioning with
```

Comment:
I remembered we were discussing that we should move the `xla_distribute_tensor` and `xla_distribute_module` implementation to the xla repo directly and only call them in the core APIs, why we didn't do this actually?

Reply:
you had a PR on `xla_distribute_tensor` about this already? https://github.com/pytorch/pytorch/pull/113214

Reply:
Yes, I moved the implementaiton already to xla, but the refactoring PR hasn't landed. Let's fix it, thanks!